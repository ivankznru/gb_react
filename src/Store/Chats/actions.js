import { ADD_CHAT, REMOVE_CHAT } from "./constants";

export const addChat = (chatId, name) => ({
  type: ADD_CHAT,
  payload: {
    id: chatId,
    name,
  },
});

export const removeChat = (chatId) => ({
  type: REMOVE_CHAT,
  payload: {
    chatId,
  },
});
