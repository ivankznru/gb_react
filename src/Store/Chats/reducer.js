import { ADD_CHAT, REMOVE_CHAT } from "./constants";

export const initialState = {
  //  chat1: {
  //    id: 'chat1',
  //     name: 'Чат 1',
  //  },
};

export default function chatsReducer(state = initialState, action) {
  switch (action.type) {
    case ADD_CHAT: {
      return {
        ...state,
        [action.payload.id]: action.payload,
      };
    }
    case REMOVE_CHAT: {
      const { chatId } = action.payload;
      const { [chatId]: chatToDelete, ...restChats } = state;
      return restChats;
    }
    default: {
      return state;
    }
  }
}
