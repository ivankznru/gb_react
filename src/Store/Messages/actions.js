import { AUTHORS } from "../../components/App/constants";
import { v4 as uuidv4 } from "uuid";
import { ADD_MESSAGE } from "./constants";

export const addMessage = (chatId, message) => ({
  type: ADD_MESSAGE,
  payload: {
    chatId,
    message,
  },
});

export const sendMessageToBot = (chatId, message) => {
  return (dispatch, getState) => {
    dispatch(addMessage(chatId, message));

    let timer = setTimeout(() => {
      dispatch(
        addMessage(chatId, {
          id: uuidv4(),
          author: AUTHORS.BOT,
          text: "Привет, я - бот!",
        })
      );

      clearTimeout(timer);
    }, 1500);
  };
};
