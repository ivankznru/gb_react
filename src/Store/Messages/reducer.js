import { ADD_MESSAGE, REMOVE_MESSAGES_BY_CHAT_ID } from "./constants";

export const initialState = {
    // chat1: [
    //     { id: 'message1', text: 'Привет', author: AUTHORS.ME },
    //     { id: 'message2', text: 'Привет!', author: AUTHORS.BOT },
    // ],
}

export default function messagesReducer(state = initialState, action) {
    switch (action.type) {
        case ADD_MESSAGE: {
            return {
                ...state,
                [action.payload.chatId]: [
                    ...(state[action.payload.chatId] || []),
                    action.payload.message,
                ],
            }
        }
        case REMOVE_MESSAGES_BY_CHAT_ID: {

            if (!state.messages.hasOwnProperty(action.payload)) {
              return state
            }
      
            const newMessages = {...state.messages};
            delete newMessages[action.payload];
      
            return {
              messages: newMessages
            }
          }
        default:
            return state
    }
}
