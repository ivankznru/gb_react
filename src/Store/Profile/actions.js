import { CHANGE_NAME, CHANGE_IS_ONLINE , CHANGE_IS_AUTHED} from "./constants";

export const changeName = (name) => ({
    type: CHANGE_NAME,
    payload: {
        name,
    },
})

export const changeIsAuthed = (isAuthed) => ({
    type: CHANGE_IS_AUTHED,
    payload: {
        isAuthed,
    },
})

export const changeNameWithThunk = (name) => {
    return (dispatch, getState) => {
        console.log(name)

        setTimeout(() => {
            dispatch(changeName(name))
        }, 1000)
    }
}

export const changeIsOnline = (isOnline) => ({
    type: CHANGE_IS_ONLINE,
    payload: {
        isOnline,
    },
})

export const changeIsOnlineWithThunk = (isOnline) => {
    return (dispatch, getState) => {
        console.log(getState())

        setTimeout(() => {
            dispatch(changeIsOnline(isOnline))
        }, 1000)
    }
}
