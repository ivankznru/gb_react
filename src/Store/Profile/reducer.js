import { CHANGE_NAME, CHANGE_IS_ONLINE } from "./constants";

const initialState = {
  name: "John",
  age: 27,
  isOnline: true,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case CHANGE_NAME: {
      return {
        ...state,
        name: action.payload.name,
      };
    }
    case CHANGE_IS_ONLINE: {
      return {
        ...state,
        isOnline: action.payload.isOnline,
      };
    }
    default:
      return state;
  }
}
