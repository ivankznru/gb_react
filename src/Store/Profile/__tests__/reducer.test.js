import { changeName } from "../actions";
import reducer from "../reducer";

describe("test profile reducer", () => {
  it("test changeName", () => {
    const received = reducer(undefined, changeName());

    expect(received).toMatchSnapshot();
  });
});
