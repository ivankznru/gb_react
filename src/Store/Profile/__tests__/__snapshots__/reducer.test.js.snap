// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile reducer test changeName 1`] = `
Object {
  "age": 27,
  "isAuthed": false,
  "isOnline": true,
  "name": undefined,
}
`;
